doc/man7/openssl-core.h.pod

```diff
@@ -69,20 +69,175 @@ It's normally passed in arrays, where the array is terminated with an
 element where all fields are zero (for non-pointers) or C<NULL> (for
 pointers).
 
-These arrays can be used both to set parameters for some object, and
-to request parameters.
+These arrays can be used both to pass parameter data to some function,
+but can also be used to request data from some function.
+For a request, this description talks about a requesting function that
+provides the C<OSSL_PARAM> array, and a responding function that fills
+in the buffers that the C<OSSL_PARAM> elements point at.
 
-C<OSSL_PARAM> is further described in L<OSSL_PARAM(3)>
+Regardless of the use, the array as well as any buffer space it points
+to must be provided by the caller (the requesting function).
+
+The fields of this structure are:
+
+=over 4
+
+=item C<key>
+
+The identity of the parameter in the form of a string.
+
+These identities are defined by the entity that provides these
+parameters, i.e. names for parameters provided by the OpenSSL
+libraries are defined by the libraries, and names for parameters
+provided by providers are defined by those providers, except for the
+pointer form of strings (see data type descriptions below)
+
+=item C<buffer>
+
+=item C<buffer_size>
+
+C<buffer> is a pointer to the memory where the parameter data is or
+shall be stored, and C<buffer_size> is its size in bytes.
+The organization of the data depends on the parameter type and flag.
+
+=item C<return_size>
+
+When an array of C<OSSL_PARAM> is used to request data, the responding
+function must set this field to indicate the actual size of the data
+stored in memory pointed at by the C<buffer> pointer, or the needed
+buffer size if the size indicated by C<buffer_size> is too small.
+
+=item C<data_type>
+
+=for comment It's still debated if this field should be present, or if
+the type should always be implied by how it's used.  Either way, these
+data types will have to be passed together with the names as an array
+of OSSL_ITEM, for discovery purposes.
+
+The C<data_type> is a value that describes the type and organization of
+the data, and can be one of:
+
+=over 4
+
+=for comment XXXX FIX ME
+
+=item C<OSSL_PARAM_INTEGER>
+
+The parameter data is an integer (signed or unsigned) of arbitrary
+length, organized in native form, i.e. most significant byte first on
+Big-Endian systems, and least significant byte first on Little-Endian
+systems.
+
+=item C<OSSL_PARAM_REAL>
+
+=for comment It's still debated if we need this or not.
+
+The parameter data is a floating point value in native form.
```

Comment:
I still believe it is needed :)

The RAND APIs will need to be provider based and I'd prefer to avoid scaled integers.